types/fh-sync.d.ts

```diff
@@ -0,0 +1,121 @@
+declare module 'fh-sync' {
+
+  export interface PendingWorkerBackoff {
+    strategy: string;
+    max: string;
+  }
+
+  export interface AckWorkerBackoff {
+    strategy: string;
+    max: string;
+  }
+
+  export interface SyncWorkerBackoff {
+    strategy: string;
+    max: number;
+  }
+
+  export interface SyncGlobalOptions {
+    pendingWorkerInterval: number;
+    pendingWorkerConcurrency: number;
+    pendingWorkerBackoff: PendingWorkerBackoff;
+    ackWorkerInterval: number;
+    ackWorkerConcurrency: number;
+    ackWorkerBackoff: AckWorkerBackoff;
+    syncWorkerInterval: number;
+    syncWorkerConcurrency: number;
+    syncWorkerBackoff: SyncWorkerBackoff;
+    schedulerInterval: number;
+    schedulerLockMaxTime: number;
+    schedulerLockName: string;
+    datasetClientUpdateConcurrency: number;
+    collectStats: boolean;
+    statsRecordsToKeep: number;
+    collectStatsInterval: number;
+    metricsInfluxdbHost?: string;
+    metricsInfluxdbPort?: number;
+    metricsReportConcurrency: number;
+    useCache: boolean;
+    queueMessagesTTL: string;
+    datasetClientCleanerCheckFrequency: string;
+  }
+
+  type StandardCb<T> = (err: Error | string | null, res: T | null) => void;
```

Comment:
Just to clarify here, because I don't think that is entirely accurate.

`variableName?` makes the variable optional in TypeScript. However, all values (even ones with explicit types) in TypeScript can be `undefined` or `null`. So for a method signature of `(name: string, age: number)`, `(null, undefined)` is perfectly valid by default.

If we are enforcing that `err` and `res` must be provided into `StandardCb` (even if they are undefined and null) then we should not make them optional. So I think we should use (err: Error | string, res: T)

The `--strictNullChecks` flag in the TypeScript 2.x compiler (not enabled by default I don't think) will stop this behaviour of `null` and `undefined` being valid values for all types.

Hope that makes sense.